File: leetcode/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> m(256, -1);
        int l = 0, r = 0, n = s.size(), len = 0;
        while (r < n) {
            if (m[s[r]] != -1) {
                l = max(m[s[r]] + 1, l);
            }
            m[s[r]] = r;
            len = max(len, r - l + 1);
            r++;
        }
        return len;
    }
};

File: neetcode-150/sliding-window/3. Longest Substring Without Repeating Characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int n = s.Length;
        int l = 0;
        int r = 0;
        int maxLen = 0;
        HashSet<char> set = new HashSet<char>();
        while (r < n) {
            char c = s[r];
            while (set.Contains(c))
            {
                set.Remove(s[l]);
                l++;
            }
            set.Add(c);
            maxLen = Math.Max(maxLen, r - l + 1);
            r++;
        }
        return maxLen;
    }
}